Source/WebCore/accessibility/AXCoreObject.h

```diff
@@ -1589,6 +1588,18 @@ void enumerateAncestors(const T& object, bool includeSelf, const F& lambda)
         enumerateAncestors(*parent, true, lambda);
 }
 
+template<typename T, typename F>
+void enumerateDescendantsIncludingIgnored(T& object, bool includeSelf, const F& lambda)
+{
+    if (includeSelf)
+        lambda(object);
+
+    for (const auto& child : object.childrenIncludingIgnored()) {
+        if (child)
```

Comment:
what allows a child to be returned from this array that is nil?

Reply:
It really never should be null...I'll remove the check and see if any tests start to fail (highly doubt it).

We wouldn't have to even consider this check if `AccessibilityChildrenVector` was `Vector<Ref<AXCoreObject>>` (which cannot be null) instead of the current `Vector<RefPtr<AXCoreObject>>`. `RefPtr` doesn't make sense — we either have an object (represented by a `Ref`), or we don't (it's not an entry in the vector).

I've been meaning to make this refactor forever, but it's a big change. Hopefully I can find the time soon.